{{ cookiecutter.repo_name }}/app/src/main/java/{{ cookiecutter.package_dir }}/utils/extension/StringExtensions.kt

```diff
@@ -0,0 +1,359 @@
+    } catch (e: ParseException) {
+        e.printStackTrace()
+    }
+
+    return convertedDate
+}
+
+/**
+ * Encode String to URL
+ */
+val String.encodeToUrlUTF8: String
+    get() {
+        return URLEncoder.encode(this, "UTF-8")
+    }
+
+/**
+ * Decode String to URL
+ */
+val String.decodeToUrlUTF8: String
+    get() {
+        return URLDecoder.decode(this, "UTF-8")
+    }
+
+/**
+ * Encode String to URL
+ */
+fun String.encodeToUrl(charSet: String = "UTF-8"): String = URLEncoder.encode(this, charSet)
+
+/**
+ * Decode String to URL
+ */
+fun String.decodeToUrl(charSet: String = "UTF-8"): String = URLDecoder.decode(this, charSet)
+
+/**
+ * Returns true if this string contains exactly the provided string.
+ * This method uses RegEx to evaluate and its case-sensitive. What makes it different from the classic
+ * [contains] is that it doesn't uses [indexOf], hence it's more performant when used on long char sequences
+ * and has much higher probabilities of not returning false positives per approximation.
+ */
+fun String.containsExact(string: String): Boolean =
+    Pattern.compile("(?<=^|[^a-zA-Z0-9])\\Q$string\\E(?=\$|[^a-zA-Z0-9])")
+        .matcher(this)
+        .find()
+
+/**
+ * Save String to a Given File
+ */
+fun String.saveToFile(file: File) = FileOutputStream(file).bufferedWriter().use {
+    it.write(this)
+    it.flush()
+    it.close()
+}
+
+/**
+ * Checks string is empty or null
+ */
+inline fun String.ifIsNullOrEmpty(action: () -> Unit) {
+    if (isNullOrEmpty()) action()
+}
+
+/**
+ * Checks string is not empty or not null
+ */
+inline fun String.ifIsNotNullOrEmpty(action: () -> Unit) {
+    if (!isNullOrEmpty()) action()
+}
+
+/**
+ * Removes all occurrences of the [value] in the string
+ * @param value A vlaue
+ * @param ignoreCase Ignore case?
+ * @return A new string with all occurrences of the [value] removed
+ */
+fun String.remove(value: String, ignoreCase: Boolean = false): String = replace(value, "", ignoreCase)
+
+/**
+ * Captialize firts letter of the string
+ */
+fun String.capitalizeFirstLetter(): String {
+    return this.substring(0, 1).uppercase() + this.substring(1)
+}
+
+/**
+ * Captialize firts letter of the each word in the string
+ */
+fun String.capitalizeFirstLetterEachWord(): String {
+    return this.lowercase()
+        .split(" ")
+        .joinToString(" ") { it.capitalize() }
+}
+
+/**
+ * Get md5 string.
+ */
+val String.md5 get() = hashAString(this, "MD5", toByteArray())
+
+/**
+ * Get sha1 string.
+ */
+val String.sha1 get() = hashAString(this, "SHA-1", toByteArray())
+
+
+fun String.removeSpace(): String = replace(" ", "")
+
+fun String.showAsToast(context: Context, duration: Int) {
+    Toast.makeText(context, this, duration).show()
+}
+
+/**
+ * Computes the substring starting at start and ending at endInclusive,
+ *      which is the length of text unless specified otherwise.
+ */
+fun String.stringSubstring(start: Int, endInclusive: Int = this.length - 1): String {
```

Comment:
This function returns substring, start indicates start of sub string and endInclusive is optional if we do not provide anything it will return subString from start index to end of string